recipes/gdcm/all/conanfile.py

```diff
@@ -25,101 +30,86 @@ class GDCMConan(ConanFile):
-            self.copy(patch["patch_file"])
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("expat/2.4.8")
-        self.requires("openjpeg/2.4.0")
+        self.requires("expat/2.4.9")
+        self.requires("openjpeg/2.5.0")
         self.requires("zlib/1.2.12")
 
     def validate(self):
-        if self.options.shared and self._is_msvc and "MT" in msvc_runtime_flag(self):
-            raise ConanInvalidConfiguration("shared gdcm can't be built with MT or MTd")
-        if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, "11")
+        if self.info.settings.compiler.cppstd:
+            check_min_cppstd(self, self._minimum_cpp_standard)
+        if is_msvc_static_runtime(self) and self.info.options.shared:
+            raise ConanInvalidConfiguration(f"{self.ref} does not support shared and static runtime together.")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
-
-    def _patch_sources(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
+        get(self, **self.conan_data["sources"][self.version],
+                  destination=self.source_folder, strip_root=True)
 
-        self._cmake = CMake(self)
-        self._cmake.definitions["GDCM_BUILD_DOCBOOK_MANPAGES"] = False
-        self._cmake.definitions["GDCM_BUILD_SHARED_LIBS"] = self.options.shared
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["GDCM_BUILD_DOCBOOK_MANPAGES"] = False
+        tc.cache_variables["GDCM_BUILD_SHARED_LIBS"] = self.info.options.shared
         # FIXME: unvendor deps https://github.com/conan-io/conan-center-index/pull/5705#discussion_r647224146
-        self._cmake.definitions["GDCM_USE_SYSTEM_EXPAT"] = True
-        self._cmake.definitions["GDCM_USE_SYSTEM_OPENJPEG"] = True
-        self._cmake.definitions["GDCM_USE_SYSTEM_ZLIB"] = True
-        if not tools.valid_min_cppstd(self, 11):
-            self._cmake.definitions["CMAKE_CXX_STANDARD"] = 11
-
-        self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
+        tc.cache_variables["GDCM_USE_SYSTEM_EXPAT"] = True
+        tc.cache_variables["GDCM_USE_SYSTEM_OPENJPEG"] = True
+        tc.cache_variables["GDCM_USE_SYSTEM_ZLIB"] = True
+        if not self.settings.compiler.cppstd:
```

Comment:
@uilianries is it possible to have same package id with compiler.cppstd=11 and without it?
I've tried it but got error
```
def package_id(self):
        if not self.info.settings.compiler.cppstd:
            self.info.settings.compiler.cppstd = self._minimum_cpp_standard
```
`gdcm/3.0.9: Invalid ID: The compiler.cppstd is not defined for this configuration`

Reply:
hm, it seems that "11" can't be even used with conan. Tried compatible packages
```
def package_id(self):
        if self.settings.compiler.cppstd == self._minimum_cpp_standard:
            compatible_pkg = self.info.clone()
            del compatible_pkg.compiler.cppstd
            self.compatible_packages.append(compatible_pkg)
```
ConanException: Invalid setting '11' is not a valid 'settings.compiler.cppstd' value.
Possible values are ['None', '14', '17', '20', '23']

Reply:
@Nekto89 

> is it possible to have same package id with compiler.cppstd=11 and without it?

No. However, you can manage the package ID: https://docs.conan.io/en/latest/creating_packages/define_abi_compatibility.html#defining-a-custom-package-id

However, customizing at this level is **forbidden** in CCI.

Reply:
> hm, it seems that "11" can't be even used with conan. Tried compatible packages

You customized your settings.yml for sure. `11` is always present there by default.

You can remove `settings.yml` and run conan create again, then Conan will generate a new settings.yml for you

Reply:
it seems that Visual Studio doesn't have C++11 as separate value for /std: option. Having CMAKE_CXX_STANDARD 11 leads to the same result as not having CMAKE_CXX_STANDARD at all.